log-collector-script/linux/kubectl-collector.sh

```diff
@@ -0,0 +1,80 @@
+#!/usr/bin/env bash
+# set -x
+export LANG="C"
+export LC_ALL="C"
+readonly CURRENT_TIME=$(date +%Y-%m-%d_%H%M-%Z)
+readonly COLLECT_DIR="kubectl-output-${CURRENT_TIME}"
+mv ${COLLECT_DIR} ${COLLECT_DIR}_bk 2>/dev/null
+mkdir -p ${COLLECT_DIR}
+
+try() {
+  local action=$*
+  echo -ne "Trying... $action "
+}
+
+ok() {
+  echo
+}
+
+pack() {
+  try "archive gathered information"
+  tar --create --verbose --gzip --file ${PWD}/eks_kubectl_commands_"${CURRENT_TIME}".tar.gz --directory="${COLLECT_DIR}" . > /dev/null 2>&1
+  ok
+}
+
+finished() {
+  echo -e "\n\t Done... your kubectl command logs are located in \n \n \t ${PWD}/eks_kubectl_commands_"${CURRENT_TIME}".tar.gz \n"
+}
+
+kubectl_get_specific_resource() {
+  object_type=$1
+  resource_name=$2
+  namespace_name=$3
+  try "kubectl get ${object_type} ${resource_name} -n ${namespace_name} -o yaml"
+  kubectl get ${object_type} ${resource_name} -n ${namespace_name} -o yaml > ${COLLECT_DIR}/kubectl-get-${object_type}-${resource_name}-${namespace_name}-${CURRENT_TIME}.yaml
+  ok
+}
+
+kubectl_get() {
+  object=$1
+  try "kubectl get ${object} --all-namespaces -o wide \n"
+  kubectl get ${object} --all-namespaces -o wide > ${COLLECT_DIR}/kubectl-get-${object}-${CURRENT_TIME}.log
+  ok
+}
+
+# object_type,resource_name,namespace. Example:pods,test-pod,kube-system
+list_get_yaml=(
+  'configmap,aws-auth,kube-system'
+  'daemonset,aws-node,kube-system'
+  'daemonset,kube-proxy,kube-system'
+  'configmap,kube-proxy,kube-system'
+  'configmap,kube-proxy-config,kube-system'
+  'deployment,coredns,kube-system'
+  'configmap,coredns,kube-system'
+)
+
+# get object for a specific resource in a namespace. Example: kubectl get configmap aws-auth -n kube-system -o yaml
+for each_get_yaml in ${list_get_yaml[@]}
+do
+  # split sub-list if available
```

Comment:
What does `if available` imply here? If there is a malformed element in `list_get_yaml`, shouldn't we at least print a warning?

Reply:
I just added `warning` to log when parsing fails.

```
Warning: Unable to parse configmap,corednskube-system. Skipped kubectl get YAML configuration for configmap,corednskube-system.
```